components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtImpl.java

```diff
@@ -199,6 +199,98 @@ public void deactivateTenant(String tenantUniqueID) throws TenantMgtException {
         }
     }
 
+    @Override
+    public void deleteTenantMetaData(String tenantUniqueIdentifier) throws TenantMgtException {
+
+        TenantManager tenantManager = TenantMgtServiceComponent.getTenantManager();
+        if (tenantManager != null) {
+            Tenant tenant = null;
+            try {
+                tenant = tenantManager.getTenant(tenantUniqueIdentifier);
```

Comment:
Once we identify the tenant id, why can't we reuse the exiting tenant delete by the tenant id method? 

What is the difference in metadata deletion?

Reply:
Yes, we can use the existing tenant deletion method by passing the tenant id, but for all tenant REST APIs we are following a standard as doing all the operation using tenant unique id. We have already implemented tenant activation/deactivation also in the same manner as we introduced a unique identifier. Thatswhy for deletion also I have followed the same approach to have consistency.

Reply:
Yes. We need to have an API to delete tenants by unique id. My question was, can't we find the tenantid from the unique id and use the old core level method to delete the tenant rather duplicating the whole code?

Reply:
Yes we can do